crates/sui-move-build/src/linters/freeze_wrapped.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Mysten Labs, Inc.
```

Comment:
I think this is fine to do over HLIR, but I think it will also work over typing? 

The typing visitors are run after expansion, so all ability information should be present. 
What issue were you anticipating with the typing visitor that caused you to reach for the HLIR?

Reply:
Thanks for explaining the subtlety here. Reverted to using typed AST.